kubernetes/metadata/config.go

```diff
@@ -58,7 +58,7 @@ func GetDefaultResourceMetadataConfig() *AddResourceMetadataConfig {
 	return &AddResourceMetadataConfig{
 		Node:       metaCfg,
 		Namespace:  metaCfg,
-		Deployment: true,
-		CronJob:    true,
+		Deployment: false,
```

Comment:
The change looks sane to me. This would need a changelog entry in the proper section as a breaking change.

Since this is a breaking change (with valid reasoning) please wait for validation from @tommyers-elastic and @mlunadia before merging.

Reply:
+1 from me

Reply:
Can you please make it at least configurable? We use `kubernetes.deployment` do distinguish between pod types while configuring elastic-agent. Hint based discovery is still limited so we rely on condition based.

Reply:
@bartoszcisek you can still enable it in your manifest 

You need to enable the kubernetes provider
```bash
data:
  agent.yml: |-
    ...
    providers.kubernetes:
      add_resource_metadata:
        deployment: true
        cronjob: true
```

I am testing with 8.11.0-SNAPSHOT
I exec in my elastic agent pod:
```bash
elastic-agent inspect -v --variables --variables-wait 2s
....
- data_stream:
    namespace: default
  id: filestream-container-logs-d105fc5a-36b3-4d5a-b97a-288d141781de-kubernetes-7107f2fe-3ae6-49fb-9aa6-5de2a5e23259.local-path-provisioner
  meta:
    package:
      name: kubernetes
      version: 1.46.0
  name: kubernetes-1
  original_id: filestream-container-logs-d105fc5a-36b3-4d5a-b97a-288d141781de
  - add_fields:
      fields:
        container:
          name: local-path-provisioner
        deployment:   < -------This is present
          name: local-path-provisioner
```

And if you need also your metrics  to be enriched,  you can change the state_pod and state_container datasets with the add_resource_metadata block

```bash
streams:
          - id: >-
              kubernetes/metrics-kubernetes.state_container-d105fc5a-36b3-4d5a-b97a-288d141781de
            data_stream:
              type: metrics
              dataset: kubernetes.state_container
           ...
            add_resource_metadata:
              deployment: true
              cronjob: true

         - id: >-
              kubernetes/metrics-kubernetes.state_pod-d105fc5a-36b3-4d5a-b97a-288d141781de
            data_stream:
              type: metrics
              dataset: kubernetes.state_pod
           ...
            add_metadata: true
            add_resource_metadata:
              deployment: true
              cronjob: true
```

There is also the option to use ingest pipelines as described in this new doc https://www.elastic.co/guide/en/fleet/master/ingest-pipeline-kubernetes.html